### PR TITLE
Add Jest as the javascript test framework

### PR DESCRIPTION
**Pull request by xuhang57**
Add jest as the javascript framework for GIJI.

https://facebook.github.io/jest/docs/tutorial-react.html#content

After pulling this PR,

try type `npm test` under the `/opt/dev/troposphere` directory and `/opt/env/troposphere` environment. 

I have written one example test case which is in `test/components/projects/common/projectListHeader.spec.js` And when you run `npm test`, that test case should pass. It is just one type of test case we could have. 

We still need to have more tests such as `snapshot tests` etc.

### CODE REVIEW

**Review by rob-baron — CHANGES_REQUESTED** [review]
Travis CI needs to work with this.

**Comment by rob-baron** [comment]
I don't like the notion of snapshot testing as it captures snapshots of serializable structures, then uses those structures in a similar manner to mock data on django.  It is better to not have this feature, this way we are sure that we are testing with the current version of the server.

**Review by rob-baron — CHANGES_REQUESTED** [review]
I was noticing that tests does not seem be run in the logs of travis.  Perhaps you need to add the line:

"npm run test" 

to the script section of the .travis.yml file.

**Comment by xuhang57** [comment]
The test won't pass because it is checking all 11 test scripts that they originally had. And those scripts are written with using another framework. I was going to delete them but I think we should keep them since they could serve as good examples about writing the unit test. 

So, I think we should do a manual check first and start writing the testing script asap.

**Comment by rob-baron** [comment]
If you want to keep the previous unit tests however, they are temporarily broken, you can skip them. This is typically done using skip as documented here (https://facebook.github.io/jest/docs/api.html).  This means we run the non-broken tests every time travis ci runs.

**Review by rob-baron — CHANGES_REQUESTED** [review]
If you want to keep the previous unit tests however, they are temporarily broken, you can skip them. This is typically done using skip as documented here (https://facebook.github.io/jest/docs/api.html). This means we run the non-broken tests every time travis ci runs.

You can also have jest ignore the 11 files or rename them so that jest doesn't find them.

**Review by rob-baron — APPROVED** [review]

File: test/components/projects/common/projectListHeader.spec.js [hunk 1]
```diff
@@ -1,33 +1,16 @@
...
 
 describe("Project List Header", function() {
-
-    beforeEach(function() {
-        title = projectCollectionFixture.length + " Projects";
+    var ReactTestUtils = require('react-addons-test-utils'),
+        ProjectListHeader = React.createFactory(require("../../../../troposphere/static/js/components/projects/common/ProjectListHeader")),
+        title;
+    it("should display the provided title", function() {
+        title = "3" + "Projects"
```
Comment (knikolla) [inline test/components/projects/common/projectListHeader.spec.js]: Remove the "3".

**Review by knikolla — CHANGES_REQUESTED** [review]

**Comment by xuhang57** [comment]
@knikolla  Thanks and removed

**Review by knikolla — APPROVED** [review]

**Pull request merged**